### DISCUSSION TITLE
Transform property name when writing header?

### DISCUSSION THREAD

**Discussion by ianrandell-sh** [JoshClose/CsvHelper#1721] [Q&A] [answered]
Hi there.
Firstly CsvHelper works great for me and does almost everything I need.  Thanks for creating it.

The last thing I need is to transform some property names when writing out the csv header.
In my case I need a csv header field to be "abc__r.xyz__c"   - note the dot.  
Obviously a c# property cannot be named as such, so I have used 4 underscores to replace it, such that the property name looks like this:
```
public class myDto
{
    public string abc__r____xyz__c { get; set; }
}
```

Is there anyway I can transform this so that in the csv it gets written out as:
```abc__r.xyz__c```

I have had a look through the code but couldn't see anything obvious.
Any ideas appreciated

**Comment by ianrandell-sh** [comment, marked as answer]
ah hang on - just found NameAttribute ....

**Reply by JoshClose** [reply]
You can can do that in a `ClassMap` as well.

**Reply by ianrandell-sh** [reply]
Ah k - thanks @JoshClose 
So using that attribute on my models works in my case.

I guess it'd maybe be _handy_ to be able to inject some kind of transformer (func) into the csvWriter.WriteHeader() call _for specific scenarios_. Eg. if you wanted to keep the csv writing part completely separate from the model's code (think custom nuget that internally uses CsvHelper).
This would mean that client code containing the models could be kept ignorant (of attributes and and typed maps).

This scenario is something I might run into in a couple of weeks when I need to break the CsvHelper-using code out.

Just a thought!

**Reply by JoshClose** [reply]
Instead of calling `WriteHeader` you can call `WriteField` and write the header by hand. Would that work?

**Reply by ianrandell-sh** [reply]
Using the overload that takes a type-converter?  Possibly it would. Will let you know if I hit the issue.
Thanks for your responsiveness 👍
